types/src/protocol_version.rs

```diff
@@ -414,4 +447,12 @@ mod tests {
         let other = ProtocolVersion::from_parts(1, 99, 99);
         assert!(current.is_compatible_with(&other));
     }
+
+    #[test]
+    fn should_serialize_to_json_properly() {
```

Comment:
I think either here or as a separate test, we should do the full round trip and deserialize from JSON back to the original.

It'd also be good to do a round trip test using `bincode` to check the non-human-readable case too.

Reply:
Do we do that for all types?

Reply:
Done in [`dbe8a60` (#926)](https://github.com/CasperLabs/casper-node/pull/926/commits/dbe8a601912d8ffb4ffb7eec8c9e7d191e92ba6e).